### PR TITLE
wrapping amqp closing with try-except

### PR DESCRIPTION
**Pull request by noakup**

### CODE REVIEW

File: cloudify/amqp_client.py [hunk 1]
```diff
@@ -101,13 +101,17 @@ def close(self):
...
-            logger.debug('Closing channel of thread {0}'.format(thread))
-            self.channel.close()
-        if self.connection:
-            logger.debug('Closing amqp connection of thread {0}'
-                         .format(thread))
-            self.connection.close()
+        try:
+            if self.channel:
+                logger.debug('Closing channel of thread {0}'.format(thread))
+                self.channel.close()
```
Comment (dankilman) [inline cloudify/amqp_client.py]: if we are already being protective like that, we may want to try/except the channel close and connection close separately. Also, we may want to debug log these cases.

Comment (noakup) [inline cloudify/amqp_client.py]: I asked Ran about logging, he saw no need but again I see what you do.

Comment (noakup) [inline cloudify/amqp_client.py]: except for logging, any other reason for trying and catching separately?

File: cloudify/amqp_client.py [hunk 2]
```diff
@@ -102,12 +102,23 @@ def close(self):
...
         if self.channel:
-            logger.debug('Closing channel of thread {0}'.format(thread))
-            self.channel.close()
+            logger.debug('Closing amqp channel of thread {0}'.format(thread))
+            try:
+                self.channel.close()
+            except Exception as e:
+                # channel might be already closed, log and continue
+                logger.debug('Failed to close amqp channel of thread {0}, '
+                             'reported error: {1}'.format(thread, e))
```
Comment (ran-z) [inline cloudify/amqp_client.py]: use str(e)

File: cloudify/amqp_client.py [hunk 3]
```diff
@@ -102,12 +102,23 @@ def close(self):
...
+
         if self.connection:
             logger.debug('Closing amqp connection of thread {0}'
                          .format(thread))
-            self.connection.close()
+            try:
+                self.connection.close()
+            except Exception as e:
+                # connection might be already closed, log and continue
+                logger.debug('Failed to close amqp connection of thread {0}, '
```
Comment (ran-z) [inline cloudify/amqp_client.py]: same

File: cloudify/amqp_client.py (hunk 2 above)
Comment (noakup) [inline cloudify/amqp_client.py]: str is done implicitly by the format command, but using repr :smile:

**Pull request merged**